### PR TITLE
Whisper tokenizer word level timestamps

### PR DESCRIPTION
**Pull request by kamilakesbi**
# What does this PR do?

This PR aims at solving issue #31778. 

When using the `_decode_asr` of Whisper Tokenizer to decode with word-level timestamps, a weird behavior is observed with tokens and timestamps going back in time. 

@xenova proposed a simple solution to solve this issue in transformers.js (see this [PR](https://github.com/xenova/transformers.js/pull/545/commits/9ac7ceb408900d4f0ab7699ef96fc498cbd4f49d)): when finding the common tokens between successive chunks, we need to add an additional step where we make sure matching tokens also have timestamps that are order. 

This solves the observed error. 

# Who can review: 

@sanchit-gandhi

### CODE REVIEW

File: tests/models/whisper/test_tokenization_whisper.py [hunk 1]
```diff
@@ -339,6 +340,130 @@ def test_basic_normalizer(self):
         self.assertEqual(decoded_output_diacritics, expected_output_diacritics)
 
 
+    def test_decode_asr_with_word_level_timestamps(self):
```
Comment (sanchit-gandhi) [inline tests/models/whisper/test_tokenization_whisper.py:343]: Could we perhaps reduce the input size to only check the timestamps from 45->50s? This should streamline the test (isolating where the bug occurs) and make it easier to confirm the correct behaviour

File: tests/models/whisper/test_tokenization_whisper.py [hunk 2]
```diff
@@ -339,6 +340,130 @@ def test_basic_normalizer(self):
...
+        # fmt: off
+        model_outputs = [
+            {
+                'stride': [30, 0, 5],
+                'tokens': torch.tensor([[
```
Comment (sanchit-gandhi) [inline tests/models/whisper/test_tokenization_whisper.py:348]: Can we do everything in numpy? The Whisper tokenizer (and it's associated tests) should be hardware agnostic (PT, JAX, TF)

**Review by sanchit-gandhi — APPROVED** [review]
Fix looks good! Just a few small comments regarding the test

File: tests/models/whisper/test_tokenization_whisper.py (hunk 1 above)
Comment (kamilakesbi) [reply tests/models/whisper/test_tokenization_whisper.py:341]: I've updated the example!

**Review by xenova — APPROVED** [review]
Thanks! 🚀 Output matches Transformers.js version now :)

**Review by amyeroberts — APPROVED** [review]
Thanks for digging into this and fixing!

**Pull request merged**